trunner/config.py

```diff
@@ -56,30 +60,74 @@ def parse_test_case(self, config):
             if key not in config:
                 raise YAMLParserError(self, f"key {key} not found in test config")
 
+    def parse_test_name(self, config):
         # Extend test name with path
         test_path = remove_prefix(str(self.path), str(PHRTOS_PROJECT_DIR) + '/')
         name = f"{test_path}/{config['name']}"
         config['name'] = name.replace('/', '.')
 
+    def parse_harness(self, config):
+        config['type'] = 'harness'
+        config['harness'] = self.path / config['harness']
+
+        if not config['harness'].exists():
+            raise YAMLParserError(self, f"harness {config['harness']} file not found")
+
+        if not str(config['harness']).endswith('.py'):
+            raise YAMLParserError(self, "harness file must has .py extension")
+
+    def parse_target(self, config):
+        config.setdefault('target', dict())
+
+        target = config['target']
+        target.setdefault('targets',
+                          [target for target in self.targets if target != 'host-pc'])
+
+        final_targets = set(target['targets'])
+        final_targets |= set(target.get('include', []))
+        final_targets -= set(target.get('exclude', []))
+
+        not_known = final_targets - set(TARGETS)
+        if not_known:
+            raise YAMLParserError(self, f"wrong target: {', '.join(map(str, not_known))}")
+
+        final_targets &= set(self.targets)
+        config['target'] = list(final_targets)
+
+    def parse_test_case(self, config):
+        self.parse_keywords(config)
+        self.parse_test_name(config)
+        self.parse_target(config)
         # If type is not known assume that it's a unit test
         config.setdefault('type', 'unit')
-        config.setdefault('ignore', False)
         config.setdefault('timeout', PYEXPECT_TIMEOUT)
 
+        config.setdefault('ignore', False)
         if not isinstance(config['ignore'], bool):
             config['ignore'] = False
 
         if config.get('harness'):
-            config['type'] = 'harness'
-            config['harness'] = self.path / config['harness']
+            self.parse_harness(config)
+
+        # Copy test per target
+        test_case = []
+        for target in config['target']:
+            test = config.copy()
+            test['target'] = target
+            test_case.append(test)
 
-            if not config['harness'].exists():
-                raise YAMLParserError(self, f"harness {config['harness']} file not found")
+        return test_case
 
-            if not str(config['harness']).endswith('.py'):
-                raise YAMLParserError(self, "harness file must has .py extension")
+    def copy_global_keywords(self, glob, local):
+        if 'targets' in local:
+            # Always take local 'targets' without copying include/exclude from global config
```

Comment:
as per my previous comment, I think selectively not-inheriting `target` property might be counter-intuitive